everett/manager.py

```diff
@@ -972,10 +992,25 @@ def __call__(self, key, namespace=None, default=NO_VALUE,
 
         # No value specified and no default, so raise an error to the user
         if raise_error:
-            raise ConfigurationMissingError(
+            msg = [
                 'namespace=%s key=%s requires a value parseable by %s' % (
-                    namespace, key, qualname(parser))
-            )
+                    namespace,
+                    key,
+                    qualname(parser)
+                )
+            ]
+
+            # Add config option doc if it exists
+            if doc:
+                msg.append(doc)
+
+            # Add config manager help doc if it exists
+            if self.doc:
+                msg.append(self.doc)
+
+            msg = '\n'.join(msg)
```

Comment:
Given that it's repeated a third time, I think it would be worth refactoring this. :)